proxy/agent.go

```diff
@@ -129,25 +151,22 @@ func (a *agent) Run(stop <-chan struct{}) {
 	// High QPS is needed to process messages on all channels.
 	rateLimiter := flowcontrol.NewTokenBucketRateLimiter(float32(1), 10)
 
-	// Set default delay to a long duration - reconciliation is a no-op in the regular case.
-	// For permanent errors, reconciliation is still attempted once after the default delay.
-	defaultDelay := 1 * time.Hour
-	delay := defaultDelay
-
 	for {
 		rateLimiter.Accept()
 
 		select {
-		case config := <-a.config:
-			// reset retry budget if and only if the desired config changes
-			if !reflect.DeepEqual(a.desired, config) {
-				a.retryBudget = a.maxRetries
-				delay = defaultDelay
-				a.desired = config
+		case config := <-a.configCh:
+			if !reflect.DeepEqual(a.desiredConfig, config) {
```

Comment:
This is exercised when a new config must be sent to envoy. Shouldn't it compare config with currentConfig, and set desiredConfig to config?